api/topo/topo.proto

```diff
@@ -31,6 +31,14 @@ service Topo {
   rpc Get (GetRequest) returns (GetResponse) {
   }
 
+  // Delete an object from topology
+  rpc Delete (DeleteRequest) returns (DeleteResponse) {
+  }
+
+  // List gets a stream of requested objects
+  rpc List (ListRequest) returns (stream ListResponse) {
```

Comment:
I'm not clear what the purpose of `List` is. We have `Get` for individual objects (unary), `Subscribe` for streamed requests, and `List` for (I'm not sure). Also, I think `ListRequest` should return `Updates` (rather than Object), as it it is long running (I presume). In topo "devices" I think we had List (with the option to subscribe) and Get - there was no separate subscribe.

BTW GUI can handle streaming responses, but not streaming requests, so what you have in `List` is perfect for it. So in summary, if `List` is there to replace `Subscribe` that would be great

Reply:
Get is for a fetching an object given its id (e.g. ls foo.txt). List is for getting a list of objects (e.g ls *.txt). It is not long running - it fetches what is currently available which is why ListRequest returns Object not Update. Watch is long running (e.g. watch -n 0.1 ls foo.txt) and returns Update since the UpdateType is available for new modifications on the store. (Note - neither List or Subscribe have streaming requests, so both should be good for onos-gui).

Reply:
onos topo get entity <id> ----> uses Get
onos topo get entities --------> uses List

Reply:
@SeanCondon - I will merge this but if you have any concerns regarding the List, then I will take it up in a subsequent PR if thats OK.

Reply:
thanks for the explanation - I understand you now